packages/target-electron/src/deltachat/webxdc.ts

```diff
@@ -89,12 +89,27 @@ export default class DCWebxdc extends SplitOut {
         }
       })
     })
+
+    /**
+     * ipcMain handler for 'open-webxdc' event invoked by the renderer
+     */
     const openWebxdc = async (
       _ev: IpcMainInvokeEvent,
       msg_id: number,
       p: DcOpenWebxdcParameters
     ) => {
-      const { webxdcInfo, chatName, displayname, addr, accountId } = p
+      const { webxdcInfo, chatName, displayname, accountId, href } = p
+      const addr = webxdcInfo.selfAddr
+      let base64EncodedHref = ''
+      const appURL = `webxdc://${accountId}.${msg_id}.webxdc`
+      if (href && href !== '') {
+        // href is user provided content, so we want to be sure it's relative
+        // relative href needs a base to construct URL
+        const url = new URL(href, 'http://dummy')
+        const relativeUrl = url.pathname + url.search + url.hash
+        // make href eval safe
+        base64EncodedHref = Buffer.from(appURL + relativeUrl).toString('base64')
```

Comment:
The current implementation looks better!

Also consider the following. I might be too paranoid and ignorant to how URLs work, but this looks more reliable to me:

```suggestion
        const url = new URL(href, 'http://dummy')

        const funalUrl = new URL(appURL)
        funalUrl.pathname = url.pathname
        funalUrl.search = url.search
        funalUrl.hash = url.hash

        // make href eval safe
        base64EncodedHref = Buffer.from(funalUrl.toString()).toString('base64')
```

Reply:
Here we rely on URL constructor to make sure noone finds a way to cheat it, since we call finally finalUrl.toString()

It is strange that the finalUrl shows 'webxdc:' as protocol but url.protocol = 'webxdc:' does not change any existing protocol and 'wbxdc://something' is not accepted as base argument. So there is some magic in the URL constructor and if that is changed it would break our implementation.

The current implementation garantuees that the locations starts with appUrl so I consider that more safe

Reply:
Yes was talking about your proposal. 
This is where URL constructor is inconsistent:
``` 
u = new URL('http://test.de')
u.toString() => 'http://test.de/'
u.procol = 'ftp:'
u.toString() => 'ftp://test.de/'
u.protocol = 'webxdc:'
u.toString() => 'ftp://test.de/'

```

Reply:
Indeed very weird.
But still, my approach guarantees that URL is a valid URL, with all the parameters properly "escaped" (if that's a term we can apply to URLs), and the base being set to a value controlled by us.
If there really is weirdness with such custom protocols, I'd also expect to see this weirdness down the line, in `window.location = url`.

If this doesn't convince you, them let's move on and resolve this thread.

Reply:
>  is a valid URL, with all the parameters properly "escaped"

that is the case also for the current implementation (as all properties come from URL), but I feel more comfortable to have control what the "origin" of the location.href is, by settings it explicit as string